src/dsp_tools/xmllib/serialise/serialise_file_value.py

```diff
@@ -0,0 +1,43 @@
+from typing import Never
+from typing import assert_never
+from typing import cast
+
+from lxml import etree
+
+from dsp_tools.xmllib.constants import DASCH_SCHEMA
+from dsp_tools.xmllib.constants import XML_NAMESPACE_MAP
+from dsp_tools.xmllib.models.config_options import Permissions
+from dsp_tools.xmllib.models.file_values import AbstractFileValue
+from dsp_tools.xmllib.models.file_values import FileValue
+from dsp_tools.xmllib.models.file_values import IIIFUri
+from dsp_tools.xmllib.value_checkers import is_string_like
+
+
+def serialise_file_value(file_value: AbstractFileValue) -> etree._Element:
+    """
+    Serialise the file value of a resource.
+
+    Args:
+        file_value: File Value
+
+    Returns:
+        File Value as etree
+    """
+    if isinstance(file_value, FileValue):
+        return _serialise_file_value(file_value, "bitstream")
+    elif isinstance(file_value, IIIFUri):
+        return _serialise_file_value(file_value, "iiif-uri")
+    else:
+        never: Never = cast(Never, file_value)
+        assert_never(never)
```

Comment:
This seems a bit odd. If you first cast the file_value to be Never, it will of course pass the next line. To me, it would seem more natural to raise an error here:

```suggestion
        raise BaseError("file_value must be either a FileValue or a IIIFUri, but you provided {class(file_value)}")
```

Reply:
That is also an option. I saw this in some code Balduin wrote and like the approach because it demonstrated that this code should actually not be reachable...

Reply:
I think it is the casting which seems odd to me. It seems more natural if you do it with a match case:

```
match file_value:
    case FileValue():
        return _serialise_file_value(file_value, "bitstream")
    case IIIFUri():
        return _serialise_file_value(file_value, "iiif-uri")
    case _:
        assert_never(file_value)
```